Userland/DevTools/Profiler/EventSerialNumber.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2021, Gunnar Beutner <gbeutner@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Assertions.h>
+#include <AK/Badge.h>
+#include <AK/NumericLimits.h>
+#include <AK/Types.h>
+
+namespace Profiler {
+
+// DistinctNumber's constructor is non-explicit which makes accidentally turning
+// an unrelated u64 into a serial number all too easy.
+class EventSerialNumber {
+public:
+    constexpr EventSerialNumber()
+    {
+    }
+
+    constexpr explicit EventSerialNumber(size_t serial)
+        : m_serial(serial)
+    {
+        VERIFY(serial == 0);
+    }
```

Comment:
Hmm, if we don't allow any number other than zero then what even is the point of taking an argument here?

Reply:
Yeah, that makes sense. I've removed the constructor and the explicit "{ 0 }" initializers. :+1: